charmcraft/const.py

```diff
@@ -40,34 +43,98 @@
 """
 
 # The minimum set of hooks to be provided for compatibility with old Juju
-MANDATORY_HOOK_NAMES = {"install", "start", "upgrade-charm"}
+MANDATORY_HOOK_NAMES = frozenset(("install", "start", "upgrade-charm"))
 HOOKS_DIRNAME = "hooks"
 
 # The minimum set of files for a charm to be considered valid
-CHARM_FILES = [
-    METADATA_FILENAME,
-    DISPATCH_FILENAME,
-    HOOKS_DIRNAME,
-]
+CHARM_FILES = frozenset(
```

Comment:
:heart:

Reply:
The only downside is some test is order sensitive when this changed to unordered, had to add a workaround.